src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoTreeNodes.tt

```diff
@@ -116,96 +116,110 @@ namespace Microsoft.SqlTools.ServiceLayer.ObjectExplorer.SmoModel
-						if (!string.IsNullOrWhiteSpace(typeToReverse))
-						{
-							WriteLine(string.Format("                   TypeToReverse = typeof({0}Querier),", typeToReverse));
-						}
-						if (!string.IsNullOrWhiteSpace(validFor))
+						XmlElement or = orNode as XmlElement;
+						if (or == null)
 						{
-							WriteLine(string.Format("                   ValidFor = {0},", GetValidForFlags(validFor)));
+							continue;
 						}
-						if (propertyValue != null && (filterValues == null || filterValues.Count == 0))
-						{
-							WriteLine(string.Format("                   Values = new List<object> {{ {0} }},", propertyValue));
-						}
-						if (filterValues != null && filterValues.Count > 0)
+
+						// Write initial declarator for the <Or> object, which is just a list of NodePropertyFilters
+						WriteLine("                filters.Add(new NodeOrFilter");
+						WriteLine("                {");
+						WriteLine("                    FilterList = new List<NodePropertyFilter> {");
+
+						foreach(var orFilterNode in or.GetElementsByTagName("Filter"))
 						{
-							string filterValueType = "object";
-							if (propertyType == "Enum")
+							XmlElement orFilter = orFilterNode as XmlElement;
+							if (orFilter == null)
 							{
-
+								continue;
 							}
-							WriteLine(string.Format("                   Values = new List<object>"));
-							WriteLine(string.Format("                   {{"));
-							for(int i = 0; i < filterValues.Count; i++)
-							{
-								string separator = "";
-								if (i != filterValues.Count - 1)
-								{
-									separator = ",";
-								}
-								var filterValue = filterValues[i];
-								WriteLine(string.Format("                      {{ {0} }}{1}", filterValue.InnerText, separator ));
 
-							}
-							WriteLine(string.Format("                   }}"));
+							// Declaration of Filter node
+							WriteLine("                        new NodePropertyFilter");
+
+							// Parse the elements in the <Filter> node into a string, and write
+							string filterNodeString = ParseFilterNode(xmlFile, type, orFilter, true /*orFilter*/);
+							Write(filterNodeString);
+
+							// Close out filter object definition
+							WriteLine("                        },");
 						}
+
+						// Close out declaration of the NodeOrFilter
+						WriteLine("                    }");
 						WriteLine("                });");
+					}
+
+					// Parse each of the top-level <Filter> nodes in <Filters>
+					foreach (var filterNode in filtersNode.SelectNodes("Filter"))
+					{
+						XmlElement filter = filterNode as XmlElement;
+						if (filter == null)
+						{
+							continue;
+						}
 
+						// Start declaration of Filter node
+						WriteLine("                filters.Add(new NodePropertyFilter");
 
+						// Parse the elements in the <Filter> node into a string, and write
+						string filterNodeString = ParseFilterNode(xmlFile, type, filter);
+						Write(filterNodeString);
```

Comment:
Given that you always write the filter nodes as soon as you parse them I'd say you can just save the extra steps and have the parse method also write it (but probably update to `ParseAndWritePropertyFilterNode`)

Reply:
Done